app/lib/common/widget/customAvatar.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:typed_data';
+import 'package:flutter/material.dart';
+import 'package:effektio/common/store/Colors.dart';
+
+class CustomAvatar extends StatelessWidget {
+  final Future<List<int>> avatar;
+  final Future<String> displayName;
+  final double radius;
+  const CustomAvatar({
+    Key? key,
+    required this.radius,
+    required this.avatar,
+    required this.displayName,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<int>>(
+      future: avatar,
+      builder: (
+        BuildContext context,
+        AsyncSnapshot<List<int>> snapshot,
+      ) {
+        if (snapshot.hasData) {
```

Comment:
the return value of `.avatar()` comes back with an option. Is this covered by this check, that it might come back without data?

Reply:
**snapshot.hasData** checks whether this snapshot contains a non-null value. I changed the code as bellows. 
```suggestion
        if (snapshot.hasData && snapshot.requireData.isNotEmpty) {
```
here, **snapshot.requireData.isNotEmpty** checks whether the required data of this snapshot is not empty.
How about this check?

Reply:
Have this code here to also handle what to do if its empty. This also fixes navigation crashes:
```
 if (snapshot.connectionState == ConnectionState.waiting) {
            return SizedBox(
              height: 20,
              width: 20,
              child: CircularProgressIndicator(color: AppColors.primaryColor),
            );
          } else {
            if (snapshot.hasData) {
              return CircleAvatar(
                radius: 25,
                backgroundImage:
                    MemoryImage(Uint8List.fromList(snapshot.requireData)),
              );
            } else {
              return CircleAvatar(
                radius: 25,
                backgroundColor: Colors.grey[700],
                child: SvgPicture.asset('assets/images/people.svg'),
              );
            }
          }
```